iroh/src/magicsock.rs

```diff
@@ -2768,18 +2904,41 @@ impl QuicMappedAddr {
         addr[1..6].copy_from_slice(&Self::ADDR_GLOBAL_ID);
         addr[6..8].copy_from_slice(&Self::ADDR_SUBNET);
 
-        let counter = ADDR_COUNTER.fetch_add(1, Ordering::Relaxed);
+        let counter = NODE_ID_ADDR_COUNTER.fetch_add(1, Ordering::Relaxed);
         addr[8..16].copy_from_slice(&counter.to_be_bytes());
 
-        Self(SocketAddr::new(IpAddr::V6(Ipv6Addr::from(addr)), 12345))
+        Self(SocketAddr::new(
+            IpAddr::V6(Ipv6Addr::from(addr)),
+            MAPPED_ADDR_PORT,
+        ))
     }
 }
 
-impl std::fmt::Display for QuicMappedAddr {
+impl TryFrom<SocketAddr> for NodeIdMappedAddr {
+    type Error = anyhow::Error;
+
+    fn try_from(value: SocketAddr) -> std::result::Result<Self, Self::Error> {
+        match value {
+            SocketAddr::V4(_) => anyhow::bail!("NodeIdMappedAddrs are all Ipv6, addr {value:?}"),
+            SocketAddr::V6(addr) => {
+                if addr.port() != MAPPED_ADDR_PORT {
+                    anyhow::bail!("not a mapped addr");
+                }
```

Comment:
I guess I have the same comments here as on the IpMappedAddr.  But I do appreciate that this **already** had a dummy port in it.  So let's not change that.  I'll see if I can remove it reasonably in a followup PR and so I'm happy to leave this as-is for now.

Reply:
made the change already and here was my logic for why it was okay to do:

We never in the code checked the port before, that was something I added with the `try_from`, so I'm comfortable taking it "away". Now we have a method `NodeIdMappedAddr::socket_addr` that returns the addr w/ the `MAPPED_ADDR_PORT` for when we need to use the socket address.